.github/workflows/ci.yml

```diff
@@ -16,7 +16,38 @@ jobs:
     steps:
     - uses: actions/checkout@v3
       with:
-        fetch-depth: 1
+        fetch-depth: 0
+    - name: Get branch name
+      id: branch-name
+      uses: tj-actions/branch-names@v6
+    - uses: nrwl/last-successful-commit-action@v1
+      id: last_successful_commit_push
+      with:
+        branch: ${{ steps.branch-name.outputs.current_branch }}
+        workflow_id: 'ci.yml'
+        github_token: ${{ secrets.GITHUB_TOKEN }}
+    - name: Get changed files
+      id: changed-files
+      uses: tj-actions/changed-files@v31
```

Comment:
More of a comment in passing, but 31 versions to get an action to detect changed-files? Seems a bit excessive, although maybe this evolved with the Actions platform and breaking changes were due to upstream changes 🤔